### PR TITLE
Use the latest MKLML release

### PR DESCRIPTION
**Pull request by ykim362**
This new mklml release contains some more optimizations for small batch sizes on SKX and KNL machines.

If you want to use dmlc's git repository, please upload the mklml file and let me know the path. Then, I will update the pull request.

Thanks.

### CODE REVIEW

File: prepare_mkl.sh [hunk 1]
```diff
@@ -57,10 +57,10 @@ MXNET_ROOT=`dirname $0`
 USE_MKLML=0
 # NOTE: if you update the following line, please also update the dockerfile at
 # tests/ci_build/Dockerfile.mkl
-VERSION_MATCH=20170210
-ARCHIVE_BASENAME=mklml_lnx_2017.0.2.20170209.tgz
+VERSION_MATCH=20170425
+ARCHIVE_BASENAME=mklml_lnx_2018.0.20170425.tgz
 MKL_CONTENT_DIR=`echo $ARCHIVE_BASENAME | rev | cut -d "." -f 2- | rev`
-MKLURL="https://github.com/dmlc/web-data/raw/master/mxnet/mklml-release/$ARCHIVE_BASENAME"
+MKLURL="https://github.com/01org/mkl-dnn/releases/download/v0.7/$ARCHIVE_BASENAME"
```
Comment (szha) [inline prepare_mkl.sh]: 0.7 seems to be a pre-release. What does it take to use the 0.9 release?

File: prepare_mkl.sh [hunk 2]
```diff
@@ -57,10 +57,10 @@ MXNET_ROOT=`dirname $0`
 USE_MKLML=0
 # NOTE: if you update the following line, please also update the dockerfile at
 # tests/ci_build/Dockerfile.mkl
-VERSION_MATCH=20170210
-ARCHIVE_BASENAME=mklml_lnx_2017.0.2.20170209.tgz
+VERSION_MATCH=20170425
+ARCHIVE_BASENAME=mklml_lnx_2018.0.20170425.tgz
```
Comment (szha) [inline prepare_mkl.sh]: The OS is still hard-coded to be linux. What does it take to enable MKLML osx and windows support, other than fixing the script?

File: prepare_mkl.sh (hunk 1 above)
Comment (ykim362) [inline prepare_mkl.sh]: That version number is not related to the mklml library itself. The library file is residing in mkl-dnn's public repo and the version number is for mkl-dnn.

Comment (szha) [inline prepare_mkl.sh]: I see. Thanks for the clarification.

File: prepare_mkl.sh (hunk 2 above)
Comment (ykim362) [inline prepare_mkl.sh]: Yes, it is mainly updating scripts. But, not just this script. We also need to edit .mk files in make directories. Then, we need to do some tests for the validation.

**Comment by ykim362** [comment]
This is merged by another pull request. So, I am closing it.
